src/components/ChatEntry.js

```diff
@@ -1,22 +1,53 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+import chatMessages from '../data/messages.json';
 
 const ChatEntry = (props) => {
+  const onHeartButtonToggle = () => {
+    console.log('Button toggled');
+    const updatedMessage = {
+      id: props.id,
+      sender: props.sender,
+      body: props.body,
+      timeStamp: props.timeStamp,
+      liked: !props.liked,
+    };
+
+    props.onUpdate(updatedMessage);
+  };
```

Comment:
So I see what you are doing here, but we've essentially created a function that does what the function in the app.js component does! We could keep the ternary operator below and simply run the onUpdate function using the prop.id.